### PR TITLE
Migrate some `app` scripts to Kotlin script

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: app/gradle/versionInfo.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    doLast {
+        versionInfoFile.writeText(versionInfoFileContents())
+    }
+}
+
+tasks.withType<KotlinCompile>().configureEach {
+    dependsOn(generateVersionInfoFile)
+}
+
+val cleanVersionInfo by tasks.registering(Delete::class) {
```
Comment (charleskorn) [inline app/gradle/versionInfo.gradle.kts:87]:
```suggestion
val cleanVersionInfo by tasks.registering<Delete> {
```

**Review by charleskorn — CHANGES_REQUESTED** [review]
One very minor suggestion - otherwise this looks great to me @sschuberth!

File: app/gradle/versionInfo.gradle.kts (hunk 1 above)
Comment (sschuberth) [reply app/gradle/versionInfo.gradle.kts:87]: Unfortunately, there's no such generic `registering` function for Kotlin DSL in Gradle, only for `register`, see https://docs.gradle.org/current/userguide/kotlin_dsl.html#using_kotlin_delegated_properties.

**Comment by charleskorn** [comment]
Thanks again @sschuberth.

**Pull request merged**